lisa/features/infiniband.py

```diff
@@ -1,13 +1,49 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
+import re
+from typing import Any, Dict, List, Tuple
 
+from assertpy import assert_that
+
+from lisa.base_tools import Sed, Uname, Wget
 from lisa.feature import Feature
+from lisa.operating_system import CentOs, Redhat, Ubuntu
+from lisa.tools import Firewall
+from lisa.tools.service import Service
+from lisa.tools.tar import Tar
+from lisa.util import LisaException
 
 FEATURE_NAME_INFINIBAND = "Infiniband"
 
 
 class Infiniband(Feature):
+
+    # Example output of ibv_devinfo:
+    # hca_id: mlx5_0
+    #     transport:                      InfiniBand (0)
+    #     fw_ver:                         16.28.4000
+    #     node_guid:                      0015:5dff:fe33:ff0c
+    #     sys_image_guid:                 0c42:a103:0065:bafe
+    #     vendor_id:                      0x02c9
+    #     vendor_part_id:                 4120
+    #     hw_ver:                         0x0
+    #     board_id:                       MT_0000000010
+    #     phys_port_cnt:                  1
+    #             port:   1
+    #                     state:                  PORT_ACTIVE (4)
+    #                     max_mtu:                4096 (5)
+    #                     active_mtu:             4096 (5)
+    #                     sm_lid:                 55
+    #                     port_lid:               693
+    #                     port_lmc:               0x00
+    #                     link_layer:             InfiniBand
+    _ib_info_pattern = re.compile(r"(\s*(?P<id>\S*):\s*(?P<value>.*)\n?)")
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.setup_rdma()
```

Comment:
Call it in `_initialize`. Let's follow the common practice, not to run long time code in constructor.